test/Feature/UpsertSpec.hs

```diff
@@ -100,30 +100,38 @@ spec =
             }
 
         it "INSERTs and ignores rows on single unique key conflict" $
-          request methodPost "/single_unique?on_conflict=unique_key" [("Prefer", "return=representation"), ("Prefer", "resolution=ignore-duplicates")]
-            [json| [
-              { "unique_key": 1, "value": "B" },
-              { "unique_key": 2, "value": "C" },
-              { "unique_key": 3, "value": "D" }
-            ]|] `shouldRespondWith` [json| [
-              { "unique_key": 3, "value": "D" }
-            ]|]
-            { matchStatus = 201
-            , matchHeaders = ["Preference-Applied" <:> "resolution=ignore-duplicates", matchContentTypeJson]
-            }
+          request methodPost "/single_unique?on_conflict=unique_key"
+              [("Prefer", "return=representation"), ("Prefer", "resolution=ignore-duplicates")]
+              [json| [
+                { "unique_key": 1, "value": "B" },
+                { "unique_key": 2, "value": "C" },
+                { "unique_key": 3, "value": "D" }
+              ]|]
+            `shouldRespondWith`
+              [json| [
+                { "unique_key": 2, "value": "C" },
+                { "unique_key": 3, "value": "D" }
```

Comment:
Hmm.. not figuring this one out. Why does it give a different result than before?

Reply:
So this tests `ignore-duplicates`. The return value should only have those rows that were actually inserted.

The difference is in the 2nd row `{ "unique_key": 2, "value": "C" }`. Before the all-rollback changes this existed already at the start of this test, now it doesn't.

A couple of tests up, there is a test `it "INSERTs and UPDATEs rows on single unique key conflict" $`. This test updates the `unique_key: 1` row, which is created in `data.sql`, and **inserts the `unique_key: 2` row**. Because those changes are rolled back, this row is now not there anymore.

We are only interested in `unique_key: 1` **not** showing up in the output here, so it's alright. I could have removed the `unique_key: 3` row alltogether, I guess...